src/clang/index.cr

```diff
@@ -21,4 +34,21 @@ class Clang::Index
   def finalize
     LibClang.dispose_index(@index)
   end
+
+  def clang_headers_dir
+    # looks for the clang's builtin headers directory in clang's search path
+    llvm_version = `llvm-config --version`.strip
+    llvm_libdir = `llvm-config --libdir`.strip
+    target_triple = Crystal::Config.default_target_triple
+    if target_triple.count('-') > 2
+      target_triple = target_triple.split('-').tap(&.delete_at(1)).join('-')
+    end
+
+    [
+      File.join("/usr/local/include/clang", llvm_version, "include"),
+      File.join("/usr/include/clang", llvm_version, "include"),
+      File.join("/usr/include", target_triple, "clang", llvm_version, "include"),
+      File.join(llvm_libdir, "clang", llvm_version, "include"),
+    ].find { |d| File.exists?(d) && File.directory?(d) }
```

Comment:
You can ask clang directly for the default include directories, instead of harcoding them. See for example:

https://github.com/ysbaddaden/clang.cr/commit/8d7d3079afc7a029e9934e0a3fc28f395cfc2026

Reply:
Thank you for the example, done !

I was thinking, if you are maintaining this clang binding (and if it's not too much work) do you think it could be used in place of the current binding ?